### PR TITLE
Config and quality of life

### PR DESCRIPTION
**Pull request by oheyjesse**
# Sssssssssssssss

This PR is for Quality Of Life fixes like better auto recompiling of JS files in dev, some basic linting setup, and a few additional little fixes.

# What do

- Add `watch` mode to webpack dev config. This watches for files, and recompiles when it sees a change. We still have to manually refresh the window - which is annoying. I was gonna go down this rabbithole but I didn't want to make too many big architectural changes at once

- Add ESLintrc - Config for ESLint - it's helpful for working in JS to have linting rules set up. I haven't gone through and formatted for this yet, though. I'd suggest we get to a point where our PR's are all in and merged up, then do a format based on the linter. 🤷‍♀ 

- Readme changes to reflect a few problems I ran into in Pythonland

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -25,6 +25,7 @@ wheels/
 *.egg
 MANIFEST
 node_modules/
+frontend/static/frontend/main.js
```
Comment (mermop) [inline .gitignore:28]: There are a couple of things like this where they're committed both here and on #18.. can you tidy that up?

File: webpack.config.js [hunk 2]
```diff
@@ -1,4 +1,8 @@
 module.exports = {
+  cache: false,
```
Comment (mermop) [inline webpack.config.js:2]: This is mostly duplicated in #18 but you've added different line 2s

File: README.md [hunk 3]
```diff
@@ -2,30 +2,49 @@
...
+
+If you only need to "build" the JS once (for example, for deployment), run:
+
+Frontend build: `npm run build`
 
-To run the server, run `python manage.py runserver`
-To compile the JS, run `npm run dev`
+If you're actively developing frontend and would like the server to "watch" for changes, you can have webpack run a dev server, to watch for changes and recompile as you go!
+
+Frontend dev server: `npm run dev`
```
Comment (mermop) [inline README.md:39]: 👍 Great documenting

**Review by mermop — APPROVED** [review]
Changes generally look fine but there's a bit of duplication with #18 - happy to merge but it might be a good idea to tidy those up!

File: .gitignore (hunk 1 above)
Comment (oheyjesse) [reply .gitignore:28]: Yep, definitely. Looks like I didn't properly separate my branches. It was a little bit of "trying to keep the concerns separate, whilst finding a few changes as I went" with regards to this one. I'll go back over and try separate them out / put the right things in the right places.

**Comment by oheyjesse** [comment]
Fixed this up - Made sure the infrastructure / webpack / linting / git / css changes are all in here. I removed the gitignore and webpack stuff from the layout PR - When I merge this, and rebase the updated master in the layout branch, those remaning "css file deleted" conflicts should well resolve fine.

**Pull request merged**